packages/hoppscotch-common/src/components/app/spotlight/index.vue

```diff
@@ -0,0 +1,220 @@
+
+const t = useI18n()
+
+const props = defineProps<{
+  show: boolean
+}>()
+
+const emit = defineEmits<{
+  (e: "hide-modal"): void
+}>()
+
+const spotlightService = useService(SpotlightService)
+
+useService(HistorySpotlightSearcherService)
+useService(UserSpotlightSearcherService)
+
+const search = ref("")
+
+const searchSession = ref<SpotlightSearchState>()
+const stopSearchSession = ref<() => void>()
+
+const scoredResults = computed(() =>
+  Object.entries(searchSession.value?.results ?? {}).sort(
+    ([, sectionA], [, sectionB]) => sectionB.avgScore - sectionA.avgScore
+  )
+)
+
+const { selectedEntry } = newUseArrowKeysForNavigation()
+
+watch(
+  () => props.show,
+  (show) => {
+    search.value = ""
+
+    if (show) {
+      const [session, onSessionEnd] =
+        spotlightService.createSearchSession(search)
+
+      searchSession.value = session.value
+      stopSearchSession.value = onSessionEnd
+    } else {
+      stopSearchSession.value?.()
+      stopSearchSession.value = undefined
+      searchSession.value = undefined
+    }
+  }
+)
+
+function runAction(searcherID: string, result: SpotlightSearcherResult) {
+  spotlightService.selectSearchResult(searcherID, result)
+  emit("hide-modal")
+}
+
+function newUseArrowKeysForNavigation() {
+  const selectedEntry = ref<[number, number]>([0, 0]) // [sectionIndex, entryIndex]
+
+  watch(search, () => {
+    selectedEntry.value = [0, 0]
+  })
+
+  const onArrowDown = () => {
+    // If no entries, do nothing
+    if (scoredResults.value.length === 0) return
+
+    const [sectionIndex, entryIndex] = selectedEntry.value
+
+    const [, section] = scoredResults.value[sectionIndex]
+
+    if (entryIndex < section.results.length - 1) {
+      selectedEntry.value = [sectionIndex, entryIndex + 1]
+    } else if (sectionIndex < scoredResults.value.length - 1) {
+      selectedEntry.value = [sectionIndex + 1, 0]
+    } else {
+      selectedEntry.value = [0, 0]
+    }
+  }
+
+  const onArrowUp = () => {
+    // If no entries, do nothing
+    if (scoredResults.value.length === 0) return
+
+    const [sectionIndex, entryIndex] = selectedEntry.value
+
+    if (entryIndex > 0) {
+      selectedEntry.value = [sectionIndex, entryIndex - 1]
+    } else if (sectionIndex > 0) {
+      const [, section] = scoredResults.value[sectionIndex - 1]
+      selectedEntry.value = [sectionIndex - 1, section.results.length - 1]
+    } else {
+      selectedEntry.value = [
+        scoredResults.value.length - 1,
+        scoredResults.value[scoredResults.value.length - 1][1].results.length -
+          1,
+      ]
+    }
+  }
+
+  function handleKeyPress(e: KeyboardEvent) {
+    if (e.key === "ArrowUp") {
+      e.preventDefault()
+      e.stopPropagation()
+
+      onArrowUp()
+    } else if (e.key === "ArrowDown") {
+      e.preventDefault()
+      e.stopPropagation()
+
+      onArrowDown()
+    } else if (e.key === "Enter") {
+      e.preventDefault()
```

Comment:
it throws an error when the result is empty and the user presses the enter key
![image](https://github.com/hoppscotch/hoppscotch/assets/53208152/1ca3bde2-1425-4287-8685-f64099ffdea8)

we can stop the execution if the scoredResults is empty
`if (scoredResults.value.length === 0) return`